sdk/nodejs/tests/runtime/tsClosureCases.ts

```diff
@@ -5017,6 +5017,322 @@ return function () { console.log(o1); console.log(o2.b.d); console.log(o3.b.d);
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+
+function __f0() {
+  return (function() {
+    with({ getAll: __getAll }) {
+
+return function () { console.log(getAll()); };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+`,
+        });
+    }
+
+    {
+        const defaultsForThing = { config: { x: "x", y: "y" } };
+        function getX() { return defaultsForThing.config.x }
+        function getAll() { const x = getX(); return { y: defaultsForThing } }
+
+        cases.push({
+            title: "Analyze property chain #27",
+            func: function () { console.log(getAll()); },
+            expectText: `exports.handler = __f0;
+
+var __defaultsForThing = {};
+var __defaultsForThing_config = {x: "x", y: "y"};
+__defaultsForThing.config = __defaultsForThing_config;
+
+function __getX() {
+  return (function() {
+    with({ defaultsForThing: __defaultsForThing, getX: __getX }) {
+
+return function /*getX*/() { return defaultsForThing.config.x; };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+
+function __getAll() {
+  return (function() {
+    with({ getX: __getX, defaultsForThing: __defaultsForThing, getAll: __getAll }) {
+
+return function /*getAll*/() { const x = getX(); return { y: defaultsForThing }; };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+
+function __f0() {
+  return (function() {
+    with({ getAll: __getAll }) {
+
+return function () { console.log(getAll()); };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+`,
+        });
+    }
+
+    {
+        const defaultsForThing = { config: { x: "x", y: "y" } };
+        function getX() { return defaultsForThing.config.x }
+        function getAll() { const x = getX(); return { y: defaultsForThing.config } }
+
+        cases.push({
+            title: "Analyze property chain #28",
+            func: function () { console.log(getAll()); },
+            expectText: `exports.handler = __f0;
+
+var __defaultsForThing = {};
+var __defaultsForThing_config = {x: "x", y: "y"};
+__defaultsForThing.config = __defaultsForThing_config;
+
+function __getX() {
+  return (function() {
+    with({ defaultsForThing: __defaultsForThing, getX: __getX }) {
+
+return function /*getX*/() { return defaultsForThing.config.x; };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+
+function __getAll() {
+  return (function() {
+    with({ getX: __getX, defaultsForThing: __defaultsForThing, getAll: __getAll }) {
+
+return function /*getAll*/() { const x = getX(); return { y: defaultsForThing.config }; };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+
+function __f0() {
+  return (function() {
+    with({ getAll: __getAll }) {
+
+return function () { console.log(getAll()); };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+`,
+        });
+    }
```

Comment:
Added several different permutations of visiting different property chains (of differing lengths), from different contexts.  Def found several issues.  But all these cases now work properly.